### PR TITLE
implement-role-repo

### PR DESCRIPTION
**Pull request by kienpham9819**
## Related Tickets:
- [#31893](https://edu-redmine.sun-asterisk.vn/issues/31893)
- [#31894](https://edu-redmine.sun-asterisk.vn/issues/31894)

## WHAT this PR does?
- implement-role repository
- implement-permission repository

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by kienpham9819** [comment]
ready

**Review by huyenttm-0901 — COMMENTED** [review]
pending

**Comment by kienpham9819** [comment]
ready

File: app/Repositories/Role/RoleRepository.php [hunk 1]
```diff
@@ -12,4 +12,15 @@ public function getModel()
     {
         return Role::class;
     }
+
+    public function update($id, $permissionIds = [])
+    {
+        $role = $this->find($id);
+        if ($role) {
+            $role->permissions()->sync($permissionIds);
+            return $role;
```
Comment (huyenttm-0901) [inline app/Repositories/Role/RoleRepository.php]: convention

File: app/Http/Controllers/RoleController.php [hunk 2]
```diff
@@ -16,15 +19,24 @@ class RoleController extends Controller
      *
      * @return \Illuminate\Http\Response
      */
-    public function __construct()
-    {
+    protected $roleRepository;
+    protected $courseRepository;
+    protected $permissionRepository;
+
+    public function __construct(RoleRepositoryInterface $roleRepository,
```
Comment (huyenttm-0901) [inline app/Http/Controllers/RoleController.php]:
```suggestion
    public function __construct(
        RoleRepositoryInterface $roleRepository,
```

**Comment by kienpham9819** [comment]
ready

**Pull request merged**